### DISCUSSION TITLE
README Code Example Error?

### DISCUSSION THREAD

**Discussion by lplawlor** [vvo/iron-session#686] [Q&A] [answered]
Perhaps I'm missing something, but it seems that the `getIronSessionData()` function in the README doesn't actually return anything, so assigning `const session = await getIronSessionData();` doesn't work. Am I right for thinking this?

```javascript
// Next.js Server Components and Server Actions (App Router)
import { cookies } from 'next/headers';
import { getIronSession } from 'iron-session';

// ↓↓↓ NO RETURN VALUE ↓↓↓
async function getIronSessionData() {
  const session = await getIronSession(cookies(), { password: "...", cookieName: "..." });
}

function Profile() {
  // ↓↓↓ ASSIGNMENT ↓↓↓
  const session = await getIronSessionData();

  return <div>{session.username}</div>;
}
```

**Comment by mcl-laura** [comment]
I have the same problem. I'm very confused. I get an empty session object if I call `getIronSession<MySessionType>(req, res, sessionOptions)`, but I get an error if I call it with cookies `getIronSession<MySessionType>(cookies(), sessionOptions)`. The error is:  node_modules\next\dist\client\components\headers.js (54:14) @ cookies
 ⨯ Invariant: cookies() expects to have requestAsyncStorage, none available.

Is nobody else having issues?

**Comment by vvo** [comment, marked as answer]
Hey there, sorry I messed up the README, it's now fixed:

```js
export async function get(req, res) {
  const session = await getIronSession(req, res, { password: "...", cookieName: "..." });
  return session;
}
```

https://github.com/vvo/iron-session
